src/OpenLoco/src/GameCommands/Vehicles/VehiclePlace.cpp

```diff
@@ -0,0 +1,299 @@
+#include "VehiclePlace.h"
+#include "Economy/Expenditures.h"
+#include "GameState.h"
+#include "Map/RoadElement.h"
+#include "Map/TileManager.h"
+#include "Map/Track/TrackData.h"
+#include "Map/TrackElement.h"
+#include "Objects/RoadExtraObject.h"
+#include "Objects/RoadObject.h"
+#include "Objects/TrackExtraObject.h"
+#include "Objects/TrackObject.h"
+#include "Vehicles/Vehicle.h"
+#include "Vehicles/VehicleManager.h"
+#include "ViewportManager.h"
+#include <OpenLoco/Core/Numerics.hpp>
+
+namespace OpenLoco::GameCommands
+{
+    static bool validateRoadPlacement(World::Pos3 pos, uint16_t trackAndDirection, const Vehicles::VehicleHead& head)
+    {
+        Vehicles::TrackAndDirection::_RoadAndDirection rtad(0, 0);
+        rtad._data = trackAndDirection;
+
+        if (rtad.isReversed())
+        {
+            rtad.setReversed(false);
+            auto& roadSize = World::TrackData::getUnkRoad(rtad._data);
+            pos -= roadSize.pos;
+            if (roadSize.rotationEnd < 12)
```

Comment:
Is 12 just std::size(World::kRotationOffset), if yes we should probably not use a magic.

Reply:
No it's something like not diagonal